lib/kazoo/topic.rb

```diff
@@ -42,5 +42,39 @@ def eql?(other)
     def hash
       [cluster, name].hash
     end
+
+    def exist?
+      stat = cluster.zk.stat(path: "/brokers/topics/#{name}")
+      stat.fetch(:stat).exists?
+    end
+
+    def create
+      result = cluster.zk.create(
+        path: "/brokers/topics/#{name}",
+        data: JSON.dump(version: 1, partitions: partition_assignment)
+      )
+
+      if result.fetch(:rc) != Zookeeper::Constants::ZOK
+        raise Kazoo::Error, "Failed to create topic #{name}. Error code: #{result.fetch(:rc)}"
+      end
+    end
+
+    def destroy
+      raise Kazoo::Error, "The topic #{name} does not exist!" unless exist?
+      result = cluster.zk.create(path: "/admin/delete_topics/#{name}")
```

Comment:
Yeah, Kafka has a watch on this ZNode, and will do the work once a child node appears. So you have to wait for the `/brokers/topics/#{name}` node to disappear after calling this.